RealmSwift/Results.swift

```diff
@@ -147,6 +147,84 @@ extension Projection: KeypathSortable {}
         return RLMIterator(collection: collection)
     }
 
+    // MARK: Flexible Sync
+
+#if swift(>=5.8)
+    /**
+     Creates a SyncSubscription matching the Results' local filter.
+
+     After committing the subscription to the realm's local subscription set, the method
+     will wait for downloads according to the `WaitForSyncMode`.
+     - see: ``WaitForSyncMode``
+
+     - Note: This method will wait for all data to be downloaded before returning when `WaitForSyncMode.always` and `.onCreation` (when the subscription is first created) is used. This requires an internet connection if no timeout is set.
+
+     __Unnamed subscriptions:__
+     If `.subscribe()` is called without a name whose query matches an unnamed subscription, another subscription is not created.
+     If `.subscribe()` is called without a name whose query matches a named subscription, an additional  unnamed subscription is created.
+     __Named Subscriptions:__
+     If `.subscribe()` is called with a name whose query matches an unnamed subscription, an additional named subscription is created.
+     __Existing name and query:__
+     If `.subscribe()` is called with a name whose name is taken on a different query, the old subscription is updated with the new query.
+     If `.subscribe()` is called with a name that's in already in use by an identical query, no new subscription is created.
+
+     - Note: This method opens a write transaction that creates or updates a subscription.
+     It's advised to *not* loop over this method in order to create multiple subscriptions.
+     This could create a performance bottleneck by opening multiple unnecessary write transactions.
+     - note To create multiple subscriptions at once use `SyncSubscription.update`.
+
+     - parameter name: The name applied to the subscription
+     - parameter waitForSync: Determines the download behavior for the subscription. Defaults to `.onCreation`.
+     See `RLMWaitForSyncMode`.
+     - parameter timeout: An optional client timeout. The client will cancel waiting for subscription downloads after this time has elapsed. Reaching this timeout doesn't imply a server error.
+     - returns: A `Results` object.
```

Comment:
Yes, the type of the function indicates that. The relevant thing to document here is that it returns `self` and not some other Results.